tools/toolchains/mbed_toolchain.py

```diff
@@ -939,6 +940,13 @@ def add_linker_defines(self):
                 self.ld.append(define_string)
                 self.flags["ld"].append(define_string)
 
+        if hasattr(self.target, 'post_binary_hook'):
+            if self.target.post_binary_hook is None:
+                define_string = self.make_ld_define(
+                    "DISABLE_POST_BINARY_HOOK", 1)
```

Comment:
Really neat idea. I want to run a slight tweak by you and see what you think.

Naming the macro `DISABLE_POST_BINARY_HOOK` implies that a post binary hook is _usually_ enabled. In general, this isn't the case, only a few targets use post binary hooks. How do you feel about changing this to macro to `POST_BINARY_HOOK_ENABLED`? So that would mean if `post_binary_hook` is not null, `POST_BINARY_HOOK_ENABLED` would be set to `1`, otherwise it'd be set to `0`?

I think this tweak makes the solution a bit more generic and potentially useful for other targets.

Reply:
The reason I named it "DISABLE_POST_BINARY_HOOK" is because for those targets that have post_binary_hook, they're actually enabled in target.json. When you override it with null (uncommon case), it's becoming a "disable". I think you might come from a different perspective where most targets do not have post_binary_hook, yes?